Build/linux/build.sh

```diff
@@ -3,69 +3,506 @@
 # Copyright(c) 2018 Intel Corporation
 # SPDX-License-Identifier: BSD-2-Clause-Patent
 
-function clean {
-	rm -R -f debug
-	rm -R -f release
-    rm -R -f ../../Bin/Debug
-    rm -R -f ../../Bin/Release
+# Fails the script if any of the commands error (Other than if and some others)
+set -e
+
+# Allow use for coloring (as long as it's within the lines)
+if [ -n "$(tput colors 2> /dev/null)" ] && [ "$(tput colors 2> /dev/null)" -ge 8 ]; then
+    bold=$(tput bold)
+    blue=$(tput setaf 12)
+    yellow=$(tput setaf 11)
+    purple=$(tput setaf 13)
+    green=$(tput setaf 2)
+    red=$(tput setaf 1)
+    reset=$(tput sgr0)
+fi
+
+# Function to print to stderr or stdout while stripping preceding tabs(spaces)
+# If tabs are desired, use cat
+print_message() {
+    if [ "$1" = "stdout" ] || [ "$1" = "-" ]; then
+        shift
+        printf "%b\\n" "${@:-Unknown Message}" | sed -e 's/^[ \t]*//'
+    else
+        printf "%b\\n" "${@:-Unknown Message}" | sed -e 's/^[ \t]*//' >&2
+    fi
+}
+
+# Convenient function to exit with message
+die() {
+    if [ -z "$*" ]; then
+        print_message "${red}Unknown error${reset}"
+    else
+        print_message "$@"
+    fi
+    print_message "The script will now exit."
+    exit 1
+}
+
+# Function for changing directory.
+cd_safe() {
+    if ! cd "$1"; then
+        shift
+        if [ -n "$*" ]; then
+            die "$@"
+        else
+            die "Failed cd to $1."
+        fi
+    fi
+}
+
+# info about self
+script_dir="$(cd "$(dirname "$0")" > /dev/null 2>&1 && pwd)"
+cd_safe "$script_dir"
+THIS_SCRIPT=$0
+
+# Help message
+echo_help() {
+    cat << EOF
+Usage: $THIS_SCRIPT [OPTION] ... -- [OPTIONS FOR CMAKE]
+-a, --all, all          Builds release and debug
+    --asm, asm=*        Set assembly compiler [$ASM]
+-b, --bindir, bindir=*  Directory to install binaries
+    --cc, cc=*          Set C compiler [$CC]
+    --cxx, cxx=*        Set CXX compiler [$CXX]
+    --clean, clean      Remove build and Bin folders
+    --debug, debug      Build debug
+    --shared, shared    Build shared libs
+-x, --static, static    Don't build shared libs
+-g, --gen, gen=*        Set CMake generator
+-i, --install, install  Install build [Default release]
+-j, --jobs, jobs=*      Set number of jobs for make/CMake [$jobs]
```

Comment:
Keep this file as simple as possible. 
1. make it easier to maintain
2. straightforward to user

Let consider remove un-related/not used ones?
-ams
-bindir
-cxx
-gen
-jobs
-prefix
-toolchain
-verbose

You can hardcore some of them instead of having the option exposed.

Reply:
Sure, although I would like to keep at least prefix and gen so that way the user can set the install prefix to be something other than /usr/local and use Ninja or some other generator if they prefer

Reply:
SvtHevcEnc.pc has prefix=/usr/local which is used by gst plugin. Let's just use the defual intall dir, in case user mess up the plugin without knowing it.

Reply:
The pkg-config is auto generated based on cmake, so if prefix is passed as `./build.sh --prefix /usr`, the script will pass `-DCMAKE_INSTALL_PREFIX=/usr` to cmake, which in turn will generate

``` bash
prefix=/usr
includedir= ${prefix}/include/svt-hevc
...
```

Reply:
thx, let me try

Reply:
Right now doing final cleanup of the script for any dangling variables

Reply:
Does using Ninja tested?

Reply:
I think I did test at least the SVT-AV1, I can test right now

Reply:
It work, although since I remove `--gen`, it can be used by `./build.sh ... -- -G"Ninja"`

Reply:
thx